typing/rec_check.ml

```diff
@@ -306,338 +322,648 @@ let classify_expression : Typedtree.expression -> sd =
         Dynamic
   in classify_expression Ident.empty
 
-let rec expression : Env.env -> Typedtree.expression -> Use.t =
-  fun env exp -> match exp.exp_desc with
+let remove_pat pat env =
+  Env.remove_list (pat_bound_idents pat) env
+
+let remove_patlist pats env =
+  List.fold_right remove_pat pats env
+
+(* Usage mode judgments.
+
+   There are two main groups of judgment functions:
+
+   - Judgments of the form "Gamma |- ... : m"
+     compute the environment of a subterm from its mode,
+     so the corresponding function has type [... -> mode -> Env.t].
+
+     We write ['a -> term_judg] in this case.
```

Comment:
Shouldn't `'a` be `...` instead, to match the convention used in the line above?